models/requests/collection_instrument.py

```diff
@@ -0,0 +1,134 @@
+from datetime import datetime
+
+
+class CollectionInstrument:
+    def __init__(
+        self,
+        survey_id,
+        language,
+        form_type,
+        title,
+        schema_version,
+        data_version,
+        ci_version=None,
+    ):
+        """
+        Constructor for CollectionInstrument object
+        :param survey_id: required, can't be None or empty
+        :param language: required, can't be None or empty
+        :param form_type: required, can't be None or empty
+        :param title: required, can't be None or empty
+        :param schema_version: required, can't be None or empty
+        :param data_version: required, can't be None or empty
+        :param ci_version: required, can't be None or empty
+        """
+        if not (survey_id and bool(survey_id.strip())):
+            raise ValueError("survey_id can't be empty or null")
+        self._survey_id = survey_id
+
+        if not (language and bool(language.strip())):
+            raise ValueError("language can't be empty or null")
+        self._language = language
+
+        if not (form_type and bool(form_type.strip())):
+            raise ValueError("form_type can't be empty or null")
+        self._form_type = form_type
+
+        if not (title and bool(title.strip())):
+            raise ValueError("title can't be empty or null")
+        self._title = title
+
+        if not (schema_version and bool(schema_version.strip())):
+            raise ValueError("schema_version can't be empty or null")
+        self._schema_version = schema_version
+
+        if not (data_version and bool(data_version.strip())):
+            raise ValueError("data_version can't be empty or null")
+        self._data_version = data_version
+
+        self._ci_version = ci_version
+        self._published_at = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+
+    @property
+    def survey_id(self):
+        return self._survey_id
```

Comment:
Replacing this class with dataclass like below, gives advantage of:
1. No need for individual `if` statement per variable (one loop in __post_init__ would suffice)
2. No need for `@property` functions for each members
3. No need for final `to_dict` function.

```suggestion
"""All different request types accepted by the application
"""
from dataclasses import dataclass


@dataclass
class CollectionInstrument:
    """CollectionInstrument object
    dataclass will validate automatically the values for null/None/empty
    """
    survey_id: str
    language: str
    form_type: str
    title: str
    schema_version: str
    data_version: str
    ci_version: str

    def __post_init__(self):
        """
        Strip spaces from around mandatory string values, and validate them
        """
        for var_name, var_type in self.__dict__['__annotations__']:
            if var_type is str:
                # Strip only of type string (not optional)
                self.__dict__[var_name] = self.__dict__[var_name].strip()
                if not self.__dict__[var_name]:
                    raise ValueError(f"{var_name} can not be spaces")
```